File: app/src/main/java/com/example/helical_conveyor_design/CrearCuenta.java
package com.example.helical_conveyor_design;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;


public class CrearCuenta extends AppCompatActivity {


    private EditText et1_creacuenta_mail, et2_crearcuenta_password,et3_crearcuenta_password2, et4_creacuente_nickName,et5_crearcuenta_profesionOficio;
    private Button btn1_crearcuenta_registrar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_crearcuenta);

        et1_creacuenta_mail = (EditText) findViewById(R.id.et_crearcuenta_mail);
        et2_crearcuenta_password = (EditText) findViewById(R.id.et_crearcuenta_password);
        et3_crearcuenta_password2 = (EditText) findViewById(R.id.et_crearcuenta_password2);
        et4_creacuente_nickName = (EditText) findViewById(R.id.et_crearcuenta_nickname);
        et5_crearcuenta_profesionOficio = (EditText) findViewById(R.id.et_crearcuenta_profesionoficio);

        btn1_crearcuenta_registrar = (Button) findViewById(R.id.btn_crearcuenta_registrar);

    }
}

File: app/src/main/java/com/example/helical_conveyor_design/ui/view/CrearCuenta.kt
package com.example.helical_conveyor_design.ui.view

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import com.example.helical_conveyor_design.databinding.ActivityCrearcuentaBinding
import com.google.firebase.auth.FirebaseAuth
import kotlinx.android.synthetic.main.activity_crearcuenta.*

class CrearCuenta : AppCompatActivity() {

    private lateinit var binding: ActivityCrearcuentaBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityCrearcuentaBinding.inflate(layoutInflater)
        setContentView(binding.root)

        registrarCuenta()
    }

    //Metodo Registrar
    private fun registrarCuenta() {

        title = "Registro"

        binding.btnCrearcuentaRegistrar.setOnClickListener {
            if (et_crearcuenta_mail.text.isNotEmpty() &&
                et_crearcuenta_password.text.isNotEmpty() &&
                et_crearcuenta_password2.text.isNotEmpty() &&
                et_crearCuenta_nickUsuario.text.isNotEmpty())

                    FirebaseAuth.getInstance()
                        .createUserWithEmailAndPassword(
                            et_crearcuenta_mail.text.toString(),
                            et_crearcuenta_password.toString()
                        ).addOnCompleteListener {
                            if (it.isSuccessful){
                                showHome(it.result?.user?.email?: "", ProviderType.BASIC)
                            }else{
                                showAlert()
                            }
                        }
        }

    }

    private fun showAlert(){
        val builder = AlertDialog.Builder(this)
        builder.setTitle("Error")
        builder.setMessage("se ha producido un error en la autenticacion al usuario")
        builder.setPositiveButton("Aceptar",null)
        val dialog: AlertDialog = builder.create()
        dialog.show()
    }

    private fun showHome (mail: String, provider: ProviderType){
        val homeIntent = Intent(this, Logueo::class.java).apply {
            putExtra("mail",mail)
            putExtra("provider", provider.name)
        }
        startActivity(homeIntent)
    }
}